src/Microsoft.AspNetCore.Mvc.Abstractions/ModelBinding/ModelStateEntry.cs

```diff
@@ -1,12 +1,14 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System.Collections.Generic;
+
 namespace Microsoft.AspNetCore.Mvc.ModelBinding
 {
     /// <summary>
     /// An entry in a <see cref="ModelStateDictionary"/>.
     /// </summary>
-    public class ModelStateEntry
+    public abstract class ModelStateEntry
```

Comment:
We are ok with this design. With the ability to mutate the ModelStateDictionary removed (since it is now `IReadOnlyDictionary`) - there isn't a scenario where a user would have to create this type.